sqlmesh/dbt/adapter.py

```diff
@@ -78,22 +79,40 @@ def quote(self, identifier: str) -> str:
 
     def dispatch(self, name: str, package: t.Optional[str] = None) -> t.Callable:
         """Returns a dialect-specific version of a macro with the given name."""
-        dialect_name = f"{self.dialect}__{name}"
-        default_name = f"default__{name}"
-
-        references_to_try = [
-            MacroReference(package=package, name=dialect_name),
-            MacroReference(package=package, name=default_name),
-        ]
+        if self.dialect:
+            references_to_try = [
+                MacroReference(package=f"{package}_{self.dialect}", name=f"{self.dialect}__{name}"),
+                MacroReference(package=package, name=f"{self.dialect}__{name}"),
+                MacroReference(package=package, name=f"default__{name}"),
+            ]
+        else:
```

Comment:
When would we not have a dialect? Target is required.

Reply:
That's a good point! I should probably just use target then